### DISCUSSION TITLE
Breaking change after 2.5.5 to 2.6.3 update?

### DISCUSSION THREAD

**Discussion by erikscheepers** [livewire/livewire#3768] [7. Bugs (archived)] [answered]
I have an issue with the latest update of Livewire. Below was working in version 2.5.5, but in 2.6.3 this is not working anymore. Is this intentionally (and how to make this work again)?

PollingTable.php:
```php
use Livewire\Component;
use App\Models\User;

class PollingTable extends Component
{
    protected $users;

    public function getUsersProperty()
    {
        return User::all();
    }
}
```
polling-table.blade.php:
```html
<div wire:poll>
    <table class="table">
        <thead>
        <tr>
            <th scope="col">#</th>
            <th scope="col">Name</th>
            <th scope="col">Email</th>
        </tr>
        </thead>
        <tbody>
        @foreach($this->users as $user)
        <tr>
            <th scope="row">{{ $user->id }}</th>
            <td>{{ $user->name }}</td>
            <td>{{ $user->email }}</td>
        </tr>
        @endforeach
        </tbody>
    </table>
</div>
```

**Comment by mabdullahsari** [comment]
I've already mentioned this [here](https://github.com/livewire/livewire/discussions/3740#discussion-3574364), with a workaround. I personally reverted to using 2.5. The current version unfortunately has many bugs and breaking changes which need fixing first.

**Comment by joshhanley** [comment, marked as answer]
@erikscheepers have a look at this PR #3640 and Caleb's comment on it https://github.com/livewire/livewire/pull/3640#issuecomment-921298407

Then can you have a look at my response to @mabdullahsari here https://github.com/livewire/livewire/discussions/3740#discussioncomment-1349614 and see if you can share your situation/ thoughts/ ideas.

Thanks!

**Reply by erikscheepers** [reply]
I understand the reason of this change. And in most of my implementations this can be fixed by using the new boot-method. But for one use-case I bump into an issue:

index.blade.php :
```html
@livewire('polling-table', ['user_id' => 2])
```
PollingTable.php :
```php
class PollingTable extends Component
{
    public $users = [];
    public $user_id;

    public function boot()
    {
        $this->users = User::all()->except($this->user_id);
    }
}
```
polling-table.blade.php :
```html
<div wire:poll>
    <table class="table">
        <thead>
        <tr>
            <th scope="col">#</th>
            <th scope="col">Name</th>
            <th scope="col">Email</th>
        </tr>
        </thead>
        <tbody>
        @foreach($users as $user)
        <tr>
            <th scope="row">{{ $user->id }}</th>
            <td>{{ $user->name }}</td>
            <td>{{ $user->email }}</td>
        </tr>
        @endforeach
        </tbody>
    </table>
</div>
```
Upon pageload you still see the user with user_id == 2, only after the first polling this user disappears. What would be the best fix to make sure the table is shown correctly when loading?

**Reply by joshhanley** [reply]
@erikscheepers in your polling table component if you assign data to the `$user_id` property (such as `2`) and it is a public property, then that data should be available with every request.

Is that not happening? If so, I might be missing something as I can't see where you use `user_id == 2` in the code your supplied.

**Reply by erikscheepers** [reply]
I supply that data this way: `@livewire('polling-table', ['user_id' => 2])`

**Reply by joshhanley** [reply]
@erikscheepers right yeah, but once the data is passed into the child component, the child component should have it's internal `$user_id` set to `2`, so then it should poll without any problems.

If it's not, then something is going wrong as that is the way it should be working. It might be because you don't have a `key` on your child component call (at least in your example above).

Where does the `2` come from in the parent? Do you have a loop that loops through users or something?

**Reply by erikscheepers** [reply]
I simplified my component to just get the issue. If you have a database table with three users (and one with id = 2), you should be able to reproduce my issue. Just copy my livewire component and put the `@livewire('polling-table', ['user_id' => 2])` on a page. Upon pageload you should see the user with id = 2; after polling, it will be gone.

In my real-life app the list of users is auto-updated by the polling. This list is shown on the edit page of an user. The current user should not be shown in the list. The edit page is generated by a default (resource) UsersController edit-method (like: `/users/2/edit`). For this example I don't need to set keys because in the boot-function the "filtering" (excluding user with id = 2) should happen.

Hope this little background helps

**Reply by joshhanley** [reply]
@erikscheepers yep, that makes sense, thanks!

And just to confirm, this worked fine on 2.5.5 and has broken since the upgrade?

When I get a chance I will have a play 🙂

Might be worth still setting a key though and seeing what it does, as it's possible morphdom isn't recognising the polling the same component and causing strange stuff to happen.

**Reply by erikscheepers** [reply]
Adding a key using version 2.6.5 of Livewire doesn't help
```html
        @foreach($users as $user)
        <tr wire:key="{{ $user->id }}">
            <th scope="row">{{ $user->id }}</th>
            <td>{{ $user->name }}</td>
            <td>{{ $user->email }}</td>
        </tr>
        @endforeach
```

Using version 2.5.5 this was my (working) code:

index.blade.php :
```html
@livewire('polling-table', ['user_id' => 2])
```
PollingTable.php :
```php
class PollingTable extends Component
{
    protected $users = [];
    public $user_id;

    public function getUsersProperty()
    {
        return User::all()->except($this->user_id);
    }
}
```

polling-table.blade.php :
```html
<div wire:poll>
    <table class="table">
        <thead>
        <tr>
            <th scope="col">#</th>
            <th scope="col">Name</th>
            <th scope="col">Email</th>
        </tr>
        </thead>
        <tbody>
        @foreach($this->users as $user)
        <tr>
            <th scope="row">{{ $user->id }}</th>
            <td>{{ $user->name }}</td>
            <td>{{ $user->email }}</td>
        </tr>
        @endforeach
        </tbody>
    </table>
</div>
```

**Reply by joshhanley** [reply]
@erikscheepers hmm ok, why have the `protected $users` property at all, couldn't you just make use of the computed property alone.

Try this:

```php
class PollingTable extends Component
{
    public $user_id;

    public function getUsersProperty()
    {
        return User::all()->except($this->user_id);
    }
}
```

```blade
<div wire:poll>
    <table class="table">
        <thead>
        <tr>
            <th scope="col">#</th>
            <th scope="col">Name</th>
            <th scope="col">Email</th>
        </tr>
        </thead>
        <tbody>
        @foreach($this->users as $user)
        <tr>
            <th scope="row">{{ $user->id }}</th>
            <td>{{ $user->name }}</td>
            <td>{{ $user->email }}</td>
        </tr>
        @endforeach
        </tbody>
    </table>
</div>
```

**Reply by erikscheepers** [reply]
Yep. That works! Thanks!

**Reply by joshhanley** [reply]
@erikscheepers awesome! Great to hear it worked 🙂
